Source/Controllers/DataControllers/SPDatabaseStructure.m

```diff
@@ -286,12 +286,13 @@ - (void)queryDbStructureWithUserInfo:(NSDictionary *)userInfo
 #endif
 
 		// Delete all stored data for the database to be updated, leaving the structure key
-		[queriedStructure removeObjectForKey:db_id];
+		[queriedStructure safeRremoveObjectForKey:db_id];
 		NSPredicate *predicate = [NSPredicate predicateWithFormat:@"NOT SELF BEGINSWITH %@", [NSString stringWithFormat:@"%@%@", db_id, SPUniqueSchemaDelimiter]];
 		[queriedStructureKeys filterUsingPredicate:predicate];
 
 		// Set up the database as an empty mutable dictionary ready for tables, and store a reference
 		[queriedStructure setObject:[NSMutableDictionary dictionary] forKey:db_id];
+		[queriedStructure safeSetObject:[NSMutableDictionary dictionary] forKey:db_id];
```

Comment:
Is it supposed to be doing both set and safeSet or should the non safe one be removed?

Reply:
um yes